### PR TITLE
FIR: Fix KT-52236 (change modality to open for overridden members)

### PR DESCRIPTION
**Pull request by sfs**
Declarations marked with `override` are implicitly `open`.

---

Also generate direct field accesses in data classes for both FIR and FE1.0.

### CODE REVIEW

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DataClassMembersGenerator.kt [hunk 1]
```diff
@@ -92,7 +92,7 @@ abstract class DataClassMembersGenerator(
             // data classes and corresponding properties can be non-final.
             // We should use getters for such properties (see KT-41284).
             val backingField = property.backingField
-            return if (!forbidDirectFieldAccess && property.modality == Modality.FINAL && backingField != null) {
+            return if (!forbidDirectFieldAccess && irClass.isFinalClass && backingField != null) {
```
Comment (mglukhikh) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DataClassMembersGenerator.kt:95]: Could you please comment this change? It influences not only K2 but also K1, so it's not so clear for me.

**Comment by mglukhikh** [comment]
Remove of FIR2IR JS test data is also not clear for me. The test data isn't identical with K1, and these tests fail in your branch.

**Comment by mglukhikh** [comment]
Also, a lot of FIR diagnostic tests fail in this branch because of `*.fir.txt` file changes.

**Comment by mglukhikh** [comment]
Also, I observe the following exception during bootstrap compilation. Looks also related. 
```
e: java.lang.IllegalStateException: None of the modifiers is found: [open]
  at org.jetbrains.kotlin.diagnostics.PositioningStrategies$modifierSetPosition$1.mark(PositioningStrategies.kt:366)
  at org.jetbrains.kotlin.diagnostics.PositioningStrategies$modifierSetPosition$1.mark(PositioningStrategies.kt:355)
  at org.jetbrains.kotlin.diagnostics.PositioningStrategy.markDiagnostic(PositioningStrategy.kt:20)
  at org.jetbrains.kotlin.diagnostics.SourceElementPositioningStrategy.markDiagnostic(SourceElementPositioningStrategy.kt:21)
  at org.jetbrains.kotlin.diagnostics.KtDiagnostic.getTextRanges(KtDiagnostic.kt:24)
  at org.jetbrains.kotlin.cli.common.fir.FirDiagnosticsCompilerResultsReporter$InFileDiagnosticsComparator.compare(FirDiagnosticsCompilerResultsReporter.kt:125)
  at org.jetbrains.kotlin.cli.common.fir.FirDiagnosticsCompilerResultsReporter$InFileDiagnosticsComparator.compare(FirDiagnosticsCompilerResultsReporter.kt:123)
  at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
  at java.util.TimSort.sort(TimSort.java:220)
  at java.util.Arrays.sort(Arrays.java:1438)
  at kotlin.collections.ArraysKt___ArraysJvmKt.sortWith(_ArraysJvm.kt:2557)
  at kotlin.collections.CollectionsKt___CollectionsKt.sortedWith(_Collections.kt:1075)
  at org.jetbrains.kotlin.cli.common.fir.FirDiagnosticsCompilerResultsReporter.reportByFile(FirDiagnosticsCompilerResultsReporter.kt:48)
  at org.jetbrains.kotlin.cli.common.fir.FirDiagnosticsCompilerResultsReporter.reportToMessageCollector(FirDiagnosticsCompilerResultsReporter.kt:25)
  at org.jetbrains.kotlin.cli.jvm.compiler.FirKotlinToJvmBytecodeCompiler.compileModule(FirKotlinToJvmBytecodeCompiler.kt:203)
  at org.jetbrains.kotlin.cli.jvm.compiler.FirKotlinToJvmBytecodeCompiler.compileModulesUsingFrontendIR(FirKotlinToJvmBytecodeCompiler.kt:142)
  at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:79)
  at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:47)
```

**Comment by sfs** [comment]
I rebased this change before submitting the PR, but apparently didn't run all the tests... that probably accounts for most of the failures. Let me run the tests locally and update the test expectations!

**Comment by sfs** [comment]
> Remove of FIR2IR JS test data is also not clear for me. The test data isn't identical with K1, and these tests fail in your branch.

Ugh, sorry about that, I know what happened. I removed all `*.fir.txt` files and then ran the tests to update them, but failed to run all tests. Do you have a better way to update test expectations for FIR?

**Comment by mglukhikh** [comment]
> Do you have a better way to update test expectations for FIR?

Looks like we don't have necessary run configuration. Probably you should now run all tests from `compiler:fir:analysis-tests` and also `Fir2IrJsTextTestGenerated`.

**Comment by mglukhikh** [comment]
@sfs , just FYI: I'm already working on test fixing, so no further actions are required. Thank you.

**Comment by sfs** [comment]
Thanks! I had already started to fix this, I just uploaded the latest state.

---

Plus there's still some diagnostic that doesn't expect override declarations to be open, which caused the bootstrap compilation failure. I still need to have a look at that anyway.

**Comment by mglukhikh** [comment]
@sfs You can look at my current version of your commit, if you wish: https://github.com/JetBrains/kotlin/commit/07363d798840c5928bef6c1a87e07b19866e29e5. I hope I fixed everything (or almost everything) there.

**Pull request closed without merging**